app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,106 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
```

Comment:
So the main idea is that subclass should not know which instance variable you use in your abstract class, they just need to implement the method which indicate how to get that variable right?  Am I doing the correct things now?